### ISSUE TITLE
Block Scripts breaks styles on vcsjones.com

### ISSUE THREAD

**Issue by jonathansampson** [brave/browser-ios#566]
**Description:**
Blocking scripts on http://vcsjones.com results in broken styles.

**Expected behavior:**
Blocking scripts should not impact styling. Exceptions to this would be styles contingent upon scripting (such as the reliance upon certain classes adding by libraries like Modernizr).

**Device:**
iPhone 6 Plus

**Brave Version:** 1.2.17 (16.11.02.15)

**Steps to reproduce:**
1. Navigate to http://vcsjones.com
2. Enable _Block Scripts_
3. Note the negative impact

**Comment by garvankeeley** [comment]
Is it possible on this site the styling is controlled by js?

**Comment by jonathansampson** [comment]
@garvankeeley That was my initial question; site-owner insists it is now.

> Kevin Jones @vcsjones
@BraveSampson there is zero JS on the site. It's just text, CSS.

https://twitter.com/vcsjones/status/797206963344044032

I will check into it more thoroughly.

**Comment by vcsjones** [comment]
>That was my initial question; site-owner insists it is not.

I would hope there isn't JavaScript on my site. In fact, there is a `default-src 'none'` CSP policy forbidding all forms of JS, external or inline.

The only thing that might be a bit unique to this CSS load is that Subresource Integrity is used, along with a CSP policy of `style-src 'self'` so that the CSS can actually load. This all works fine when Scripts are allowed, but does not when Scripts are disabled.

**Comment by jonathansampson** [comment]
@vcsjones I'm assuming that might be the appropriate area to investigate, as a cautionary check of the site didn't turn up any other script resources.

**Comment by vcsjones** [comment]
Actually, it breaks images, too, which you can see on this page.

https://vcsjones.com/2016/11/03/building-a-nas/

Strange!

**Comment by vcsjones** [comment]
So I tracked this down to the `sandbox` CSP directive. If the `sandbox` directive is present, then it appears all external assets, images, css, etc, are blocked by the "Block Scripts" option.

**Comment by garvankeeley** [comment]
We block js by:
• not loading network resources that are javascript
• setting X-WebKit-CSP = "script-src none" on the response header to the webview

Any ideas what we should do differently?

**Comment by vcsjones** [comment]
In a later build of Brave, `1.2.18`, the issue is a little more severe. Now, clicking the Brave icon that opens the right-side drawer doesn't work at all. This only occurs on `vcsjones.com` - the drawer works correctly on other sites. The address bar also doesn't display the domain, it always displays "Search or enter address". I can split this in to a separate issue if desired.

**Comment by garvankeeley** [comment]
omg. that is horrid, I am filing it now.

**Comment by vcsjones** [comment]
>Any ideas what we should do differently?

Unfortunately not at the moment, I'll see if I can figure anything out though.
